packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      ComponentClass: Component.extend({
+        classNames: 'testing123'
+      }),
+      template: '<p>hello {{yield}}</p>'
+    });
 
-  equal(jQuery('div.testing123', '#qunit-fixture').text(), 'hello world', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('div.testing123').text().trim();
+      assert.equal(text, 'hello world', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Late-registered components can be rendered with custom `layout` property'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>there goes {{my-hero}}</div>');
+    this.addComponent('my-hero', {
+      ComponentClass: Component.extend({
+        classNames: 'testing123',
+        layout: this.compile('watch him as he GOES')
+      })
+    });
 
-QUnit.test('Late-registered components can be rendered with custom `layout` property', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>there goes {{my-hero}}</div>'));
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'there goes watch him as he GOES', 'The component is composed correctly');
+    });
+  }
 
-  boot(() => {
-    appInstance.register('component:my-hero', Component.extend({
-      classNames: 'testing123',
-      layout: compile('watch him as he GOES')
-    }));
-  });
+  ['@test Late-registered components can be rendered with template registered on the container'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>hello world {{sally-rutherford}}-{{#sally-rutherford}}!!!{{/sally-rutherford}}</div>');
+    this.addComponent('sally-rutherford', {
+      ComponentClass: Component,
+      template: 'funkytowny{{yield}}'
+    });
 
-  equal(jQuery('#wrapper').text(), 'there goes watch him as he GOES', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'hello world funkytowny-funkytowny!!!', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Late-registered components can be rendered with template registered on the container', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>hello world {{sally-rutherford}}-{{#sally-rutherford}}!!!{{/sally-rutherford}}</div>'));
+  ['@test Late-registered components can be rendered with ONLY the template registered on the container'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>hello world {{borf-snorlax}}-{{#borf-snorlax}}!!!{{/borf-snorlax}}</div>');
 
-  boot(() => {
-    appInstance.register('template:components/sally-rutherford', compile('funkytowny{{yield}}'));
-    appInstance.register('component:sally-rutherford', Component);
-  });
+    this.addComponent('borf-snorlax', {
```

Comment:
per the original test, this should be registered onto an autobooted application instance